src/internal/methods/NeoEsp32LcdXMethod.h

```diff
@@ -0,0 +1,551 @@
+            LCD_CAM.lcd_user.lcd_cmd = 0;            // No command at LCD start
+            // Dummy phase(s) MUST be enabled for DMA to trigger reliably.
+
+            // Set up DMA descriptor list (length and data are set before xfer)
+            desc = (dma_descriptor_t *)LcdBuffer; // At start of alloc'd buffer
+            for (int i = 0; i < num_desc; i++) {
+                desc[i].dw0.owner = DMA_DESCRIPTOR_BUFFER_OWNER_DMA;
+                desc[i].dw0.suc_eof = 0;
+                desc[i].next = &desc[i + 1];
+            }
+            desc[num_desc - 1].dw0.suc_eof = 1;
+            desc[num_desc - 1].next = NULL;
+
+            // Alloc DMA channel & connect it to LCD periph
+            gdma_channel_alloc_config_t dma_chan_config = {
+                .sibling_chan = NULL,
+                .direction = GDMA_CHANNEL_DIRECTION_TX,
+                .flags = {.reserve_sibling = 0}};
+            gdma_new_channel(&dma_chan_config, &dma_chan);
+            gdma_connect(dma_chan, GDMA_MAKE_TRIGGER(GDMA_TRIG_PERIPH_LCD, 0));
+            gdma_strategy_config_t strategy_config = {.owner_check = false,
+                                                        .auto_update_desc = false};
+            gdma_apply_strategy(dma_chan, &strategy_config);
+
+            // Enable DMA transfer callback
+            gdma_tx_event_callbacks_t tx_cbs = {.on_trans_eof = dma_callback};
+            gdma_register_tx_event_callbacks(dma_chan, &tx_cbs, NULL);
+        }
+    }
+
+    void Destruct()
+    {
+        if (LcdBuffer == nullptr)
+        {
+            return;
+        }
+
+        // TODO: destroy peripheral pin settings
+        // i2sSetPins(busNumber, -1, -1, -1, false);
+        // i2sDeinit(busNumber);
+
+        gdma_reset(dma_chan);
+
+        heap_caps_free(LcdBuffer);
+
+        LcdBufferSize = 0;
+        LcdBuffer = nullptr;
+
+        MuxMap.Reset();
+    }
+
+    void ResetBuffer()
+    {
+        // to keep the inner loops for EncodeIntoDma smaller
+        // they will just OR in their values
+        // so the buffer must be cleared first
+        if (MuxMap.IsNoMuxBusesUpdate())
+        {
+            // clear all the data in preperation for each mux channel to add
+            memset(LcdBuffer, 0x00, LcdBufferSize);
+        }
+    }
+
+    void FillBuffer(uint8_t** dmaBuffer,
+            const uint8_t* data, 
+            size_t sizeData, 
+            uint8_t muxId)
+    {
+        MuxMap.EncodeIntoDma(dmaBuffer,
+            data,
+            sizeData,
+            muxId);
+    }
+
+    void StartWrite()
+    {
+        if (MuxMap.IsAllMuxBusesUpdated())
+        {
+            MuxMap.ResetMuxBusesUpdated();
+            
+            gdma_reset(dma_chan);
+            LCD_CAM.lcd_user.lcd_dout = 1;
+            LCD_CAM.lcd_user.lcd_update = 1;
+            LCD_CAM.lcd_misc.lcd_afifo_reset = 1;
+
+            uint32_t xfer_size = MuxMap.MaxBusDataSize * DmaBytesPerPixelByte;
```

Comment:
Entire descriptors should be set up here since earlier memset clears the entire LcdBuffer.